### DISCUSSION TITLE
Support key code aliases

### DISCUSSION THREAD

**Discussion by dedebenui** [nickcoutsos/keymap-editor#154] [Ideas]
I'm not using the standard US layout, so I copied [QMK's aliases](https://github.com/qmk/qmk_firmware/tree/master/quantum/keymap_extras) and adapted them for ZMK. Here is an example:
```
#define CH_AGRV QUOTE // à
```

Would it be feasable to parse those and have them available in the editor? having `CH_AGRV` displayed would be fine, but it would be amazing if it could get the `à` from the comment

**Comment by nickcoutsos** [comment]
Hi @dedebenui, it isn't something I do dynamically but a common approach for ZMK users is to include [zmk-locale-generator](https://github.com/joelspadin/zmk-locale-generator) headers in their keymap, which the app will detect and use that extended list for keycode selection and rendering.

From what I can tell that's the Swiss mapping but I'm not sure if it's the DE or FR version. Both are available, so you can download the appropriate header file from zmk-locale-generator's releases, copy it to your `config/` folder and include it locally with `#include "keys_XXX.h"`. Naming is different so if you're already referencing the QMK aliases you'd need to update that.

**Comment by dedebenui** [comment]
Oh wow, how did I miss that? Thanks for the tip, I had no idea this existed. I guess that solves my problem, thank you!

**Reply by nickcoutsos** [reply]
Not your fault! I don't actually list it as a feature because I don't know first hand if it's enough support for using other locales. When I added it I had been seeing more people using headers from zmk-locale-generator and wanted to avoid users encountering errors.

Anyway, glad it's working out!
